homeassistant/components/mqtt/device_tracker/schema_yaml.py

```diff
@@ -26,13 +33,34 @@
 )
 
 
-async def async_setup_scanner_from_yaml(hass, config, async_see, discovery_info=None):
+async def async_setup_scanner_from_yaml(
+    hass: HomeAssistant, config, async_see, discovery_info=None
+):
     """Set up the MQTT tracker."""
     devices = config[CONF_DEVICES]
     qos = config[CONF_QOS]
     payload_home = config[CONF_PAYLOAD_HOME]
     payload_not_home = config[CONF_PAYLOAD_NOT_HOME]
     source_type = config.get(CONF_SOURCE_TYPE)
+    config_entry = hass.config_entries.async_entries(mqtt.DOMAIN)[0]
+    subscriptions: list[Callable] = []
+
+    hass.data[MQTT_DATA_DEVICE_TRACKER_LEGACY] = {
```

Comment:
It's better to use a dataclass instead of a dict when needing to store multiple items under the same key in hass.data. We can then easily type the items.

Reply:
Addressed by https://github.com/home-assistant/core/pull/75482